src/server/scripts/EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp

```diff
@@ -93,7 +91,7 @@ class npc_shadowfang_prisoner : public CreatureScript
         InstanceScript* instance = creature->GetInstanceScript();
 
         if (instance && instance->GetData(TYPE_FREE_NPC) != DONE && instance->GetData(TYPE_RETHILGORE) == DONE)
-            player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, GOSSIP_ITEM_DOOR, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+            player->ADD_GOSSIP_ITEM_DB(Player::GetDefaultGossipMenuForSource(creature), 0, GOSSIP_SENDER_MAIN, GOSSIP_OPTION_GOSSIP);
```

Comment:
Very nice, my bad for not reading the details on that function. :-p Just compiled a new test now.
Going to update this PR if it works as expected. :smirk_cat: